Src/main.c

```diff
@@ -79,7 +79,11 @@ int main(void)
   /* Reset of all peripherals, Initializes the Flash interface and the Systick. */
   HAL_Init();
 
+
   /* USER CODE BEGIN Init */
+  HAL_ADC_Start(&hadc);
+  HAL_TIM_PWM_Init(&htim16);
+  HAL_TIM_PWM_Start(&htim16, TIM_CHANNEL_1);
```

Comment:
1. You should do this after we initialize all of the peripherals (lines 96-101) cause right now you are starting them, but the objects don't really exist yet
2. You don't need to call HAL_TIM_PWM_Init() since we already initialize the timer.